### PR TITLE
app-backup/deja-dup: re-add package

### PR DESCRIPTION
**Pull request by oz123**
Once, this useful package was part of the ebuild tree.
I updated the dependencies and bumped to EAPI 7.
I'm adopting it as a proxy maintainer.

Signed-off-by: Oz N Tiram <oz.tiram@gmail.com>

### CODE REVIEW

File: app-backup/deja-dup/deja-dup-42.7.ebuild [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	gnome-base/dconf
+	gnome-base/gvfs[fuse]
+"
+DEPEND="${COMMON_DEPEND}
+	$(vala_depend)
+	virtual/pkgconfig
+	dev-util/intltool
+	dev-util/itstool
+	sys-devel/gettext
+	test? ( >=dev-libs/appstream-glib-0.7.16 )
```
Comment (juippis) [inline app-backup/deja-dup/deja-dup-42.7.ebuild:40]: I'd say these are all `BDEPEND` with EAPI-7. This also allows you to remove `COMMON_DEPEND` from the ebuild. Define the first block as `DEPEND` and inherit that in `RDEPEND`.

File: app-backup/deja-dup/deja-dup-42.7.ebuild [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+"
+
+src_prepare() {
+	vala_src_prepare
+	eapply_user
```
Comment (juippis) [inline app-backup/deja-dup/deja-dup-42.7.ebuild:45]: We usually call `default` as the first thing in `src_prepare`.

File: app-backup/deja-dup/deja-dup-42.7.ebuild [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit meson eutils gnome2-utils vala xdg
```
Comment (juippis) [inline app-backup/deja-dup/deja-dup-42.7.ebuild:6]: *eutils.eclass* is not used in here.

File: app-backup/deja-dup/deja-dup-42.7.ebuild [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+DESCRIPTION="Simple backup tool using duplicity back-end"
+HOMEPAGE="https://gitlab.gnome.org/World/deja-dup"
+SRC_URI="https://gitlab.gnome.org/World/${PN}/-/archive/${PV}/${P}.tar.bz2"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
```
Comment (juippis) [inline app-backup/deja-dup/deja-dup-42.7.ebuild:14]: Please don't add prefix keywords to new ebuilds. Did you test on x86?

File: app-backup/deja-dup/deja-dup-42.7.ebuild [hunk 5]
```diff
@@ -0,0 +1,56 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit meson eutils gnome2-utils vala xdg
+
+DESCRIPTION="Simple backup tool using duplicity back-end"
+HOMEPAGE="https://gitlab.gnome.org/World/deja-dup"
```
Comment (juippis) [inline app-backup/deja-dup/deja-dup-42.7.ebuild:9]: Add https://wiki.gnome.org/Apps/DejaDup as the primary homepage, keeping the gitlab page as 2ndary homepage.

File: app-backup/deja-dup/deja-dup-42.7.ebuild [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+SRC_URI="https://gitlab.gnome.org/World/${PN}/-/archive/${PV}/${P}.tar.bz2"
+
+LICENSE="GPL-3"
```
Comment (juippis) [inline app-backup/deja-dup/deja-dup-42.7.ebuild:12]:
```suggestion
LICENSE="GPL-3+"
```

File: app-backup/deja-dup/deja-dup-42.7.ebuild [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+COMMON_DEPEND="
+	>=app-crypt/libsecret-0.18.6[vala]
+	>=dev-libs/glib-2.64:2[dbus]
+	>=dev-libs/json-glib-1.2
+	dev-libs/libgpg-error
+	>=gui-libs/libhandy-1.0
+	>=net-libs/libsoup-2.48
+	>=x11-libs/gtk+-3.22:3
+	>=x11-libs/libnotify-0.7
+	>=app-backup/duplicity-0.7.14
```
Comment (juippis) [inline app-backup/deja-dup/deja-dup-42.7.ebuild:27]: Sort (place app-backup/duplicity above everything else).

File: app-backup/deja-dup/deja-dup-42.7.ebuild [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+RDEPEND="${COMMON_DEPEND}
+	gnome-base/dconf
+	gnome-base/gvfs[fuse]
+"
+DEPEND="${COMMON_DEPEND}
+	$(vala_depend)
+	virtual/pkgconfig
+	dev-util/intltool
+	dev-util/itstool
+	sys-devel/gettext
```
Comment (juippis) [inline app-backup/deja-dup/deja-dup-42.7.ebuild:39]: Sort (place virtual/pkgconfig below everything in this selected block)

**Comment by scardracs** [comment]
@oz123 ping

File: app-backup/deja-dup/deja-dup-42.7.ebuild (hunk 5 above)
Comment (oz123) [reply app-backup/deja-dup/deja-dup-42.7.ebuild]: I assume you refer to https://projects.gentoo.org/qa/policy-guide/ebuild-format.html#pg0104, I removed this.
Is this OK?

File: app-backup/deja-dup/deja-dup-42.7.ebuild (hunk 6 above)
Comment (oz123) [reply app-backup/deja-dup/deja-dup-42.7.ebuild]: done.

File: app-backup/deja-dup/deja-dup-42.7.ebuild (hunk 3 above)
Comment (oz123) [reply app-backup/deja-dup/deja-dup-42.7.ebuild:6]: removed.

File: app-backup/deja-dup/deja-dup-42.7.ebuild (hunk 2 above)
Comment (oz123) [reply app-backup/deja-dup/deja-dup-42.7.ebuild:45]: I added this too.

File: app-backup/deja-dup/deja-dup-42.7.ebuild (hunk 4 above)
Comment (oz123) [reply app-backup/deja-dup/deja-dup-42.7.ebuild]: No, this is a mistake. I just added those as my ebuild is based on the older ebuild.

File: app-backup/deja-dup/deja-dup-42.7.ebuild (hunk 7 above)
Comment (oz123) [reply app-backup/deja-dup/deja-dup-42.7.ebuild]: done.

File: app-backup/deja-dup/deja-dup-42.7.ebuild (hunk 8 above)
Comment (oz123) [reply app-backup/deja-dup/deja-dup-42.7.ebuild]: done.

File: app-backup/deja-dup/deja-dup-42.7.ebuild (hunk 5 above)
Comment (scardracs) [reply app-backup/deja-dup/deja-dup-42.7.ebuild]: @oz123 I think he mean something like HOMEPAGE="https://wiki.gnome.org/Apps/DejaDup https://gitlab.gnome.org/World/deja-dup"

**Comment by oz123** [comment]
@scardracs I am caught in other projects at the moment. Sorry for the delay.

File: app-backup/deja-dup/deja-dup-42.7.ebuild (hunk 1 above)
Comment (oz123) [reply app-backup/deja-dup/deja-dup-42.7.ebuild]: Still need to do this, sorry.

**Review by juippis — APPROVED** [review]
Thanks!

**Pull request closed without merging**